Framework/Beamline/src/SpectrumDefinition.cpp

```diff
@@ -0,0 +1,30 @@
+#include "MantidBeamline/SpectrumDefinition.h"
+
+#include <algorithm>
+
+namespace Mantid {
+namespace Beamline {
+
+/// Returns the size of the SpectrumDefinition, i.e., the number of detectors
+/// (or rather detector positions) that the spectrum comprises.
+size_t SpectrumDefinition::size() const { return m_data.size(); }
+
+/// Returns a const reference to the pair of detector index and time index at
+/// the given `index` in the spectrum definition.
+const std::pair<size_t, size_t> &SpectrumDefinition::
+operator[](const size_t index) const {
+  return m_data[index];
+}
+
+/// Adds a pair of detector index and time index to the spectrum definition. The
+/// time index defaults to zero when not specified.
+void SpectrumDefinition::add(const size_t detectorIndex,
+                             const size_t timeIndex) {
+  auto index = std::make_pair(detectorIndex, timeIndex);
```

Comment:
I am copying an example for clarification that we discussed on Slack (short answer: we do not want to prevent different time indices):

Say we want a spectrum that covers a small region along the x axis, marked by `xxx`:

Static case:
```
time   x --->
0      0 1 2 3 4
           xxx
```
Detectors 2 and 3 are part of the spectrum (time index not needed, access `DetectorInfo.l2(2)` and `DetectorInfo.l2(3)`.


Scanning case:

```
time   x --->
0      0 1 2 3 4
1       0 1 2 3 4
2        0 1 2 3 4
           xxx
```
All in all, five detector positions (of 3 different detectors) contribute. Access, e.g., `DetectorInfo.l2(2,0)` or `DetectorInfo(1,2)`.

So in the former case, `SpectrumDefinition` contains `{{2,0},{3,0}}`. In the latter case it contains `{{2,0},{3,0}},{2,1},{1,2},{2,2}}`.